File: Anul II/Semestrul II/Ingineria sistemelor software/MA/src/main/java/ro/iss/ma/controller/LoginController.java
package ro.iss.ma.controller;

import ro.iss.ma.domain.Angajat;
import ro.iss.ma.domain.Sef;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.control.*;
import javafx.fxml.FXML;
import javafx.scene.layout.AnchorPane;
import javafx.stage.Modality;
import javafx.stage.Stage;
import ro.iss.ma.service.Service;

import java.io.IOException;
import java.time.LocalDateTime;
import java.util.Optional;

public class LoginController {

    @FXML
    private TextField unTextField;
    @FXML
    private PasswordField psTextField;

    private Service service;
    private Stage stage;

    @FXML
    private void initialize() {
    }

    public LoginController( ) {
    }

    public void setService(Service service) {

        this.service = service;
    }

    public void setStage(Stage stage) {
        this.stage = stage;
    }
    private void clearFields() {
        unTextField.setText("");
        psTextField.setText("");

        unTextField.setPromptText("username");
        psTextField.setPromptText("password");
    }

    @FXML
    public void handleLogin() {
        String username = unTextField.getText();
        String password = psTextField.getText();

        if (service.loginAngajat(username, password)){
            Optional<Angajat> userOptional = service.findOneA(username);
            Optional<Sef> userOptional2 = service.findOneS(username);
            if (userOptional.isPresent()) {
                Angajat user = userOptional.get();
                showAngajatWindow(user, service);
            }
            else if (userOptional2.isPresent()){
                Sef user = userOptional2.get();
                showSefWindow(user, service);
            }
            else{
                MessageAlert.showMessage(null, Alert.AlertType.INFORMATION, "Invalid", "Utilizatorul sau parola sunt incorecte.");
            }
        }
    }

    public static void showSefWindow(Sef u, Service service) {
        try {
            FXMLLoader loader = new FXMLLoader();
            loader.setLocation(LoginController.class.getResource("/sef.fxml"));

            AnchorPane root = loader.load();

            Stage requestStage = new Stage();
            requestStage.setTitle(u.getNume()+" "+u.getPrenume());
            requestStage.initModality(Modality.WINDOW_MODAL);
            Scene scene = new Scene(root);
            requestStage.setScene(scene);

            SefController angajatController = loader.getController();
            angajatController.setService(service, requestStage);
            angajatController.setSef(u);

            requestStage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void showAngajatWindow(Angajat u, Service service) {
        try {
            FXMLLoader loader = new FXMLLoader();
            loader.setLocation(LoginController.class.getResource("/angajat.fxml"));

            AnchorPane root = loader.load();

            Stage requestStage = new Stage();
            requestStage.setTitle(u.getNume()+" "+u.getPrenume());
            requestStage.initModality(Modality.WINDOW_MODAL);
            Scene scene = new Scene(root);
            requestStage.setScene(scene);

            AngajatController angajatController = loader.getController();
            angajatController.setService(service, requestStage);
            u.setIntrare(LocalDateTime.now());
            service.updateAngajatI(u);
            angajatController.setAngajat(u);
            requestStage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}

File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruriC/controller/LoginController.cs
using System.IO;
using System.Windows.Markup;
using AgentiiZboruriC.domain;
using AgentiiZboruriC.service;

namespace AgentiiZboruriC.controller;

using System;
using System.Windows;
using System.Windows.Controls;

public class LoginController
{
    private TextBox UnTextField;
    private PasswordBox PsTextField;
    private Button Login;
    private AngajatService service;
    private BiletService biletService;
    private ZborService zborService;
    private Window dialogStage;
    private Angajat angajat;

    public void Initialize(Window window)
    {
        // Căutați controalele în arborele vizual al ferestrei
        UnTextField = (TextBox)window.FindName("UnTextField");
        PsTextField = (PasswordBox)window.FindName("PsTextField");
        Login = (Button)window.FindName("Login");

        // Atribuiți handlerul de evenimente butonului Login
        Login.Click += HandleLogin;
    }

    public void SetService(AngajatService service, BiletService biletService, ZborService zborService, Window stage)
    {
        this.service = service;
        this.biletService = biletService;
        this.zborService = zborService;
        this.dialogStage = stage;
        
         
        Initialize(stage);

    }

    private void ClearFields()
    {
        UnTextField.Text = "";
        PsTextField.Password = "";
    }

    public void HandleLogin(object sender, RoutedEventArgs e)
    {
        string username = UnTextField.Text;
        string password = PsTextField.Password;

        if (service.Login(username, password))
        {
            Angajat user = service.FindOneU(username);
            if (user != null)
            {
                ShowPage(user);
                dialogStage.Close();
            }
            else
            {
                MessageAlert.ShowMessage(null, MessageBoxImage.Information, "Invalid", "Utilizatorul sau parola sunt incorecte.");
            }
        }
    }

    private void ShowPage(Angajat u)
    {
        try
        {
            // Load the XAML file
            using (var stream = new FileStream("/angajat.xaml", FileMode.Open))
            {
                var root = (Window)XamlReader.Load(stream);

                Window requestStage = new Window();
                requestStage.Title = u.ToString();
                requestStage.Content = root;

                AngajatController angajatController = (AngajatController)root.DataContext;
                angajatController.SetService(service, biletService, zborService, requestStage);
                angajatController.SetAngajat(u);

                requestStage.Show();
            }
        }
        catch (IOException e)
        {
            Console.WriteLine(e);
        }
    }
}